### DISCUSSION TITLE
Node 20.x defaults to pnpm v6 which leads to turbo-ignore errors

### DISCUSSION THREAD

**Discussion by mmachatschek** [vercel/community#5175] [Bug Report]
### Page to Investigate

https://vercel.com/docs/deployments/builds/package-managers

### Repro
1. Create a Vercel Project which uses Node 20.x beta
2. Use custom `"ignoreCommand": "pnpm dlx turbo-ignore"`
3. See error in build log

### Issue

pnpm v8.3.1 added a fixed incompatibilities with node >= 20 https://github.com/pnpm/pnpm/issues/6424
The default version of pnpm on vercel is v6 which has known issues and incompatibilities with node 20.

- I tried to set the `"packageManager": "pnpm@8.12.1"` flag in the package.json of the app which is not being respected by the build environment for the `ignoreCommand`

![image](https://github.com/vercel/community/assets/10237069/9d7d45e2-ae77-4684-b546-d3929672ddd0)

- I tried to update pnpm itself but that also fails because of the node incompatibility issue:

![image](https://github.com/vercel/community/assets/10237069/7438ccfc-0a07-42c9-8d02-51191e93263f)

**Comment by TooTallNate** [comment]
Thanks for bringing this up. We are looking into it.

In the meantime you can use `npx turbo-ignore` as the Ignore command. Note that pnpm 8 should already be properly used in the other commands - this issue is specific to the Ignore command.

**Reply by mmachatschek** [reply]
I tested `npx turbo-ignore` which works (using it as a workaround). I was under the assumption that the turbo-ignore command needs to be executed from the correct package manager in order to calculate the right dependency graph.

> Note that pnpm 8 should already be properly used in the other commands - this issue is specific to the Ignore command.

Yes. the other commands (install, build) properly install and infer the expected `packageManager` which is defined in the `package.json`

**Comment by ofhouse** [comment]
Happy New Year and thanks for your patience!

We're back from the holidays and just shipped a change which sets the package manager version in the [Ignored Build Step](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel) in the same way it's set for the other commands.

So the pnpm version you see in `ignoreCommand` should now match the one that is used in the `install` or `build` command.
Happy Coding!

**Reply by mmachatschek** [reply]
that's great news. Thanks felix
